cmd/tugger/main.go

```diff
@@ -246,45 +298,40 @@ func validateAdmissionReviewHandler(w http.ResponseWriter, r *http.Request) {
 
 	ar := v1beta1.AdmissionReview{}
 	if err := json.Unmarshal(data, &ar); err != nil {
-		log.Println(err)
+		log.WithError(err).WithField("body", string(data)).Error("could not parse request")
 		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
 
 	namespace := ar.Request.Namespace
-	log.Printf("AdmissionReview Namespace is: %s", namespace)
+	log.Debugf("AdmissionReview Namespace is: %s", namespace)
 
 	admissionResponse := v1beta1.AdmissionResponse{Allowed: true}
 	if !contains(whitelistedNamespaces, namespace) {
 		pod := v1.Pod{}
 		if err := json.Unmarshal(ar.Request.Object.Raw, &pod); err != nil {
-			log.Println(err)
+			log.WithError(err).WithField("object", ar.Request.Object.Raw).Error("could unmarshal pod spec")
 			w.WriteHeader(http.StatusBadRequest)
 			return
 		}
 
-		// Handle containers
-		for _, container := range pod.Spec.Containers {
-			log.Println("Container Image is", container.Image)
-
-			if !containsRegisty(whitelistedRegistries, container.Image) {
-				message := fmt.Sprintf("Image is not being pulled from Private Registry: %s", container.Image)
-				log.Printf(message)
-				SendSlackNotification(message)
-				admissionResponse.Allowed = false
-				admissionResponse.Result = getInvalidContainerResponse(message)
-				goto done
-			} else {
-				log.Printf("Image is being pulled from Private Registry: %s", container.Image)
-				admissionResponse.Allowed = true && admissionResponse.Allowed
+		var validateImage func(string) bool
+		if policy != nil {
```

Comment:
I think there's an edge case for a policy file without rules in which case the existing logic in ValidateImage will return false for every input. 
```suggestion
		if policy != nil  && len(policy.Rules) > 0 {
```

Reply:
The helm chart implementation currently protects against this configuration, but it's reasonable to assume it could be deployed without helm...

If we want to consider a policy with no rules as invalid, I think this logic should go in `Load()`, where it validates the config, not here in the webhook handler.

However, I'm not convinced that a policy with no rules is invalid. It would be a clever way to block pod creation in a cluster for a bit, if you had a need to do that. Maybe `Load()` should just log a warning if there are no rules, since it's likely a misconfiguration.

Reply:
Added validation to `Load()` in 85bb981.